non-critical-infra/modules/hydra-queue-builder-v2.nix

```
@@ -0,0 +1,274 @@
+            cfg.pingInterval
+            "--speed-factor"
+            cfg.speedFactor
+            "--max-jobs"
+            cfg.maxJobs
+            "--tmp-avail-threshold"
+            cfg.tmpAvailThreshold
+            "--store-avail-threshold"
+            cfg.storeAvailThreshold
+            "--load1-threshold"
+            cfg.load1Threshold
+            "--cpu-psi-threshold"
+            cfg.cpuPsiThreshold
+            "--mem-psi-threshold"
+            cfg.memPsiThreshold
+          ]
+          ++ lib.optionals (cfg.ioPsiThreshold != null) [
+            "--io-psi-threshold"
+            cfg.ioPsiThreshold
+          ]
+          ++ (builtins.concatMap (v: [
+            "--supported-features"
+            v
+          ]) cfg.supportedFeatures)
+          ++ (builtins.concatMap (v: [
+            "--mandatory-features"
+            v
+          ]) cfg.mandatoryFeatures)
+          ++ lib.optionals (cfg.useSubstitutes != null) [
+            "--use-substitutes"
+          ]
+          ++ lib.optionals (cfg.mtls != null) [
+            "--server-root-ca-cert-path"
+            cfg.mtls.serverRootCaCertPath
+            "--client-cert-path"
+            cfg.mtls.clientCertPath
+            "--client-key-path"
+            cfg.mtls.clientKeyPath
+            "--domain-name"
+            cfg.mtls.domainName
+          ]
+        );
+
+        User = "hydra-queue-builder";
+        Group = "hydra";
+
+        ReadWritePaths = [
+          "/nix/var/nix/gcroots/"
+          "/nix/var/nix/daemon-socket/socket"
+        ];
+        ReadOnlyPaths = [ "/nix/" ];
+        RuntimeDirectory = "hydra-queue-builder-v2";
+
+        PrivateNetwork = false;
+        SystemCallFilter = [
+          "@system-service"
+          "~@privileged"
+          "~@resources"
+        ];
+
+        ProtectSystem = "strict";
+        ProtectHome = true;
+        PrivateTmp = true;
+        PrivateDevices = true;
+        ProtectKernelTunables = true;
+        ProtectControlGroups = true;
+        RestrictSUIDSGID = true;
+        PrivateMounts = true;
+        RemoveIPC = true;
+        UMask = "0077";
+
+        CapabilityBoundingSet = "";
+        NoNewPrivileges = true;
+
+        ProtectKernelModules = true;
+        SystemCallArchitectures = "native";
+        ProtectKernelLogs = true;
+        ProtectClock = true;
+
+        RestrictAddressFamilies = "AF_UNIX AF_INET AF_INET6";
+
+        LockPersonality = true;
+        ProtectHostname = true;
+        RestrictRealtime = true;
+        MemoryDenyWriteExecute = true;
+        PrivateUsers = true;
+        RestrictNamespaces = true;
+      };
+    };
+    systemd.tmpfiles.rules = [
+      "d /nix/var/nix/gcroots/per-user/hydra-queue-builder 0755 hydra-queue-builder hydra -"
+    ];
+    nix = {
+      settings = {
+        allowed-users = [ "hydra-queue-builder" ];
+        trusted-users = [ "hydra-queue-builder" ];
+      };
+      extraOptions = ''
+        experimental-features = nix-command
+      '';
```

```suggestion
        allowed-users = [ "hydra-queue-builder" ];
        experimental-features = [ "nix-command" ];
        trusted-users = [ "hydra-queue-builder" ];
      };
```
otherwise this won't be merged properly